crates/common/mqtt_channel/Cargo.toml

```diff
@@ -9,6 +9,7 @@ rust-version = "1.58.1"
 
 [dependencies]
 async-trait = "0.1"
+bytes = "1.3.0"
```

Comment:
I would avoid to add an explicit dependency to this crate that is used under the hood by `rumqttc`.

And in this case, this is useless since the payload of of last will message only needs to `impl Into<Vec<u8>>`

The line `message: Bytes::from(payload.into()),` can simply written `message: payload` since the payload  `impl Into<Vec<u8>>`.

Reply:
The `LastWill` message is

 pub struct LastWill {
    pub topic: String,
    pub message: Bytes,
    pub qos: QoS,
    pub retain: bool,
}

But message is

Reply:
Fixed